docs/docs/development/module-creation.md

```diff
@@ -0,0 +1,171 @@
+| `snippets/`      | ❌          | ❌           | ❌         | ❌       | ✅        | ❌    |
+
+The below table reminds of the purpose of each of these files and folders, if you are not already familiar with them:
+
+| File or Folder   | Description                                                                            |
+| ---------------- | -------------------------------------------------------------------------------------- |
+| `board/`         | This folder contains definitions for boards and shields.                               |
+| `dts/`           | This folder contains devicetree bindings and includes.                                 |
+| `CMakeLists.txt` | CMake file for the module                                                              |
+| `Kconfig`        | Kconfig file for the module                                                            |
+| `include/`       | Folder to put C header files into                                                      |
+| `src/`           | Folder for C source files                                                              |
+| `snippets/`      | Folder for [snippets](https://docs.zephyrproject.org/latest/build/snippets/index.html) |
+
+Note that the `include` and `src` folders are not mandated by the module system, and all of these can be positioned anywhere in your module's filetree if you adjust the `zephyr/module.yml` accordingly.
+
+## Inclusion in ZMK Modules Repository
+
+:::warning
+The ZMK modules repository is currently experimental, and the below content is subject to change. ZMK modules is also missing tooling.
+:::
+
+Once you have written and tested your module, you should prepare it for inclusion into `zmk-modules`. The primary reasons for you wanting to do so are:
+
+- It makes discovering and using your module significantly easier for other users
+- Automated tools can help you maintain your module notifying you if there are any potentially breaking changes that you should investigate or new features that you may want to account for.
+
+### Requirements
+
+The following requirements are necessary for a module to be included in `zmk-modules`:
+
+- [ ] The module is named according to convention
+- [ ] The module has been tested and confirmed working
+- [ ] The module features a `README.md` file describing the module
+- [ ] The module is licensed under an MIT-compatible license, preferably through a `LICENSE` file
+- [ ] The module follows the general style of other modules of the same type found in `zmk-modules`
+- [ ] There are no unused files or folders present in the module
+- [ ] The module is up-to-date with the current version of ZMK
+- [ ] If the module defines keyboards or hardware components, said keyboards/components are available for public use (sold by a vendor, open source, etc.)
+- [ ] If the module defines boards or shields, they follow the "good board/shield definition practices"
```

Comment:
```suggestion
- [ ] If the module defines boards or shields, they follow the board/shield definition practices as recommended in the ZMK docs
```
How about this, there are some things Zephyr recommends that we would recommend against, such as enabling the console by default.

Reply:
That makes sense, I was thinking of board stuff which we don't have.